### DISCUSSION TITLE
Adding Single Entry to Firewall Alias List

### DISCUSSION THREAD

**Discussion by 1hamidreza** [jaredhendrickson13/pfsense-api#510] [General]
Hi, 

In API v1 we had an endpoint : 
/api/v1/firewall/alias/entry

and we could add a new entry to the current Alias list without removing or changing the previous entries in the alias. 

In API v2 seems to be no option for adding entry and PATCH method for endpoint : 
/api/v2/firewall/alias

would replace the current Alias list and if you want to add a new entry you would include all of the current entries in your request to have them all in the Alias list after adding.
Any idea how to do that ?

**Comment by jaredhendrickson13** [comment]
Currently, v2 does not support appending new values to an existing array directly. This limitation is primarily due to how v2 defines schemas, which mandates that endpoints interact with entire objects rather than individual or primitive field values. 

As of now, to add new alias address entries, you can follow these steps:

1. Perform a GET request to retrieve the current values of your alias.
2. In your code, take the `address` array obtained from the GET request and append your new value(s).
3. Submit a PATCH request to update the array with all the values.

I'm certainly not opposed to adding a new [control parameter](https://pfrest.org/COMMON_CONTROL_PARAMETERS/) that would allow a PATCH request to append the submitted array values to the existing array instead of replacing it. I am still accepting new features for v2.1.0 if you'd like to submit a [feature request](https://github.com/jaredhendrickson13/pfsense-api/issues/new?assignees=&labels=backlog%2C+feature+request&projects=&template=feature-request.md) for this functionality.

Thank you for your input!

**Reply by 1hamidreza** [reply]
Thanks for the reply. Will raise a feature request as this seems good feature for remote firewall triggering.
